app/models/spree/ability.rb

```diff
@@ -191,6 +191,8 @@ def add_product_management_abilities(user)
         OpenFoodNetwork::Permissions.new(user).managed_product_enterprises.include? product.supplier
       end
 
+      can [:admin, :index], :products_v3
```

Comment:
I have to admit I need to check the docs each time I look at the ability file! I don't understand what `:admin` is for, but tried it out and can see that this works for the `index` action. 

I also don't understand how the existing products page is working and why we need to add this line! The existing rule above (`can`...`:index`...`Spree::Product`) is validated on a specific `product`, but when showing an index there's no single product to validate, and we manage access in the query instead.

In any case, I can see that this works 👍

Reply:
I must admit that it's not totally clear to me as well. Specially this `:admin`. But I understand why I should add this line. 
This is due to 

https://github.com/openfoodfoundation/openfoodnetwork/blob/cb4aa946de96de017e5e6a361ee952f265c90a3a/app/controllers/spree/admin/base_controller.rb#L45-L57

when requesting `ProductsV3Controller`, we go through the `else`, and then `record` is finally equal to `:products_v3`